tests/integration/install/test_installation.py

```diff
@@ -409,3 +415,50 @@ def test_installation_with_dependency_upload(ws, installation_ctx, mocker):
 
     installation_ctx.deployed_workflows.repair_run("failing")
     assert installation_ctx.deployed_workflows.validate_step("failing")
+
+
+@pytest.fixture
+def no_connection_ctx(env_or_skip, make_random) -> MockInstallationContext:
+    """Create a context without internet connection.
+
+    Resources are not required as the lack of internet connection prohibits creation.
+    """
+
+    @credentials_strategy("raise_connection_error", [])
+    def raise_connection_error(_: Any):
+        """Mock no internet access by raising a ConnectionError"""
+
+        def inner():
+            raise RequestsConnectionError("no internet")
+
+        return inner
+
+    config = Config(credentials_strategy=raise_connection_error, retry_timeout_seconds=1)
+    ws = WorkspaceClient(host=env_or_skip("DATABRICKS_HOST"), config=config)
+    ctx = MockInstallationContext(
```

Comment:
this is a unit test (uses mocks), not an integration test (uses fixtures). move.

Reply:
I agree it make sense now to move this to the unit tests as it (now) works without internet connection. Note that all integration tests using the `installation_ctx` fixture are using the `MockInstallationContext`